tests/modules/vault-secrets.nix

```diff
@@ -0,0 +1,129 @@
+{ nixosPath, self, ... }@args:
+
+(import "${nixosPath}/tests/make-test-python.nix" ({ pkgs, ... }:
+  let
+    ssh-keys = import "${self.inputs.nixpkgs}/nixos/tests/ssh-keys.nix" pkgs;
+    vault-port = 8200;
+    vault-address = "http://server:${toString vault-port}";
+  in rec {
+    name = "vault-secrets";
+    nodes = {
+      server = { pkgs, ... }:
+        let
+          serverArgs =
+            "-dev -dev-root-token-id='root' -dev-listen-address='0.0.0.0:${toString vault-port}'";
+        in {
+          # An unsealed dummy vault
+          networking.firewall.allowedTCPPorts = [ vault-port ];
+          systemd.services.dummy-vault = {
+            wantedBy = [ "multi-user.target" ];
+            path = with pkgs; [ getent vault ];
+            script = "vault server ${serverArgs}";
```

Comment:
Why wrap it in an unnecessary shell process when you could just `ExecStart` directly?

Reply:
Because `path` only applies to `script`